File: input_read_keyboard_basic.py
# Reading basic keyboard state

import harfang as hg

hg.InputInit()

while True:
	state = hg.ReadKeyboard('raw')  # note: the 'raw' device can be queried without an open window, use 'default' otherwise

	for key in range(hg.K_Last):
		if state.Key(key):
			print('Key down: %d' % key)

	if state.Key(hg.K_Escape):
		break

File: input_read_keyboard_basic.lua
-- Reading basic keyboard state

hg = require("harfang")

hg.InputInit()

while true do
	state = hg.ReadKeyboard('raw')  -- note: the 'raw' device can be queried without an open window, use 'default' otherwise

	for key=0, hg.K_Last-1 do
		if state:Key(key) then
			print(string.format('Key down: %d' , key))
		end
	end
	if state:Key(hg.K_Escape) then
		break
	end
end